internal/lm/nvml.go

```diff
@@ -218,6 +224,45 @@ func newGPUModeLabeler(devices []resource.Device) (Labeler, error) {
 	return labels, nil
 }
 
+func newImexDomainLabeler(devices []resource.Device) (Labeler, error) {
+	var commonClusterUUID, commonCliqueID string
+
+	for _, d := range devices {
+		clusterUuid, err := d.GetClusterUUID()
+		if err != nil {
+			return nil, err
+		}
+		if commonClusterUUID == "" {
+			commonClusterUUID = clusterUuid
+		}
+		if commonClusterUUID != clusterUuid {
+			klog.Warningf("cluster UUID mismatch: %s != %s", commonClusterUUID, clusterUuid)
+			return nil, fmt.Errorf("cluster UUID mismatch: %s != %s", commonClusterUUID, clusterUuid)
```

Comment:
Won't returning an error here, trigger GFD itself to error out?

Reply:
you are right, a nil return is enough, we are logging the warning.

Reply:
PR now on new direction